python/demo/gmsh/demo_gmsh.py

```diff
@@ -5,33 +5,101 @@
 # SPDX-License-Identifier:    LGPL-3.0-or-later
 #
 # =====================================
-# Mesh generation using Gmsh and pygmsh
+# Mesh generation using the GMSH python API
 # =====================================
 
-import numpy as np
-import pygmsh
+from dolfinx.mesh import create_mesh, create_meshtags
+from dolfinx.io import XDMFFile, ufl_mesh_from_gmsh
+from dolfinx.cpp.io import perm_gmsh, extract_local_entities
+from dolfinx import cpp
 from mpi4py import MPI
+import gmsh
+import numpy as np
+
+
+def gather_gmsh_physical_entities():
+    """
+    Convenience function that extracts all entitites with
+    physical tags, creating mesh data structures for each
+    cell type.
+    Returns a dictionary where the key is the gmsh cell type
+    and its content is a dictionary of cells and cell data.
+
+    """
+    phys_grps = gmsh.model.getPhysicalGroups()
+    meshes = {}
+    for dim, tag in phys_grps:
+        entities = gmsh.model.getEntitiesForPhysicalGroup(dim, tag)
+        for entity in entities:
+            element_types, elem_tags, node_tags =\
```

Comment:
As a followup to Garth comment. I believe that these two lines can fit in a single one (we have the limit at 120 chars, right?)